docs/nidigital/class.rst

```diff
@@ -3384,6 +3384,18 @@ channels
         passes a string of :python:`'0, 1, 2'` to the set attribute function.
 
 
+pins
+----
+
+    .. py:attribute:: nidigital.Session.pins[]
+
+        .. code:: python
+
+            session.pins['0-2'].channel_enabled = True
```

Comment:
In Digital driver, pin names cannot begin with a number. Also, range doesn't make sense for pins.

The example should show something like `'PinA,PinB'`

This example does bring up one thing.. Since many methods that take in `channel_list` can accept both channel numbers and pins, doing something like `pins['0-2']` will work. Similarly, doing `channels['PinA']` will also work. I wonder if we should put some checks in Python to prevent such confusing misuse.

Reply:
We do not do any of these checks in other drivers either. Now that our internal metadata is getting more complete, this might be possible, but would require changes to the metadata and to how we generate the code. So something like this works as well:
``` python
with nifgen.Session('dev1') as session:
    session.channel['marker1'].some_trigger = 'dev2/clk_out'
```

Reply:
First, I want to point out that in nidigital C API, there are functions that accept "channelList" (e.g. niDigital_TDR) param - These functions actually accept both channel numbers and pin lists.
 
Then there are functions that accept only pinList (e.g. niDigital_ConfigureTimeSetDriveFormat) param - These functions accepts only pin lists.
 
There are no functions that accept only channel numbers.

> We do not do any of these checks in other drivers either. Now that our internal metadata is getting more complete, this might be possible, but would require changes to the metadata and to how we generate the code. So something like this works as well:
>```
>with nifgen.Session('dev1') as session:
>    session.channel['marker1'].some_trigger = 'dev2/clk_out'
>```

But the difference is, in nifgen, driver will return an error for `channel['marker1']`. But in nidigital, `pins['0-2']` and `channels['PinA']` will both work for methods that accept channel_list parameter.
 
Also, I think we need changes to codegen and/or metadata to make pins rep_cap work for methods that accept `channel_list` i.e. both channel numbers and pin lists.

Reply:
> The example should show something like 'PinA,PinB'

Or the more pythonic ['PinA', 'PinB].